server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/WatcherThread.java

```diff
@@ -45,25 +45,29 @@
 
   @Override
   public void run() {
-    boolean stopped = false;
-    while (!stopped) {
-      try {
+    try {
+      boolean stopped = false;
+      while (!stopped) {
         processRef.getProcess().waitFor();
         askedForRestart = processRef.getCommands().askedForRestart();
         processRef.getCommands().acknowledgeAskForRestart();
-
-        // finalize status of ProcessRef
-        processRef.stop();
-
-        // terminate all other processes, but in another thread
-        monitor.stopAsync();
+        askForStop();
         stopped = true;
-      } catch (InterruptedException ignored) {
-        // continue to watch process
       }
+    } catch (InterruptedException ignored) {
+      askForStop();
+      Thread.currentThread().interrupt();
```

Comment:
call to `Thread.currentThread().interrupt()` is useless since we dont free the thread and continue to loop
new code doesn't ensure anymore the watched thread is stopped
old code is correct